### PR TITLE
Fixes vertical rhythm

### LINKED ISSUE #177: Fix site-wide vertical rhythm

### PR DESCRIPTION
**Pull request by pitchinvasion**
Obviously a lot of changes here, it's not perfect and I imagine samm might want to do a few tweaks but we've managed to get it fitting a vertical grid wherever possible (vertical heights and vertical centring plus occasionally needing to have half values for leading meant it was impossible to get it perfect but we just used the grid as a guide).

We also need to do more on this - fixing the about page and removing modular-scale margins but we can do more on it later

We're merging this into the form styling branch (#205) so we can carry on working on this, but obviously need some input to this.

Closes #177

### CODE REVIEW

File: source/rhythm.html.haml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+%section{style: "padding:0;margin:0;"}
```
Comment (sjmog) [inline source/rhythm.html.haml]: Let's jsut call this 'typography.html.haml'?

File: source/sass/_mixins.scss [hunk 2]
```diff
@@ -69,7 +66,28 @@
...
+  margin-bottom: $base-spacing;
   padding-left: $base-spacing;
   text-align: left;
 }
+
+// Generates correct vertical rhythm for header elements
+//
+// Step is how many steps on the modular scale
+// so if you want modular-scale(3) as the font size, pass in 3
+@mixin ma-header($step, $margin-lines: 1) {
```
Comment (sjmog) [inline source/sass/_mixins.scss]: Is `$margin-lines` something about the number of 'lines of vertical rhythm'? I feel like `$rhythm-lines` or `$rhythm-increments` is a better name

**Comment by sjmog** [comment]
Checking it out live:

- This is just so much better. Everything feels 'together', such great work.
- I'd love to see this with the rhythm set to 1.618. I see in `variables.scss` we have this line:

```scss
$modular-scale-ratio: $perfect-fourth;
```

but I don't know where to find the other variables (a comment here with options please?)

- The 'Apply' button at the bottom of each section has the wrong vertical padding. It should be taller than that.

@pitchinvasion

**Comment by pitchinvasion** [comment]
Thanks for comments samm. I'll add in those comments and a few more when I get a chance. And yeah we tried it with golden ratio, I actually think header sizes do use modular scale but obviously that scale is much larger with golden ratio which makes large sizes and gives us a little less control over sizes, but if we can get it to work I'm all in favour.

Another thing to try might be instead to set line-height to $golden instead, that looked pretty good when we tried it

**Comment by pitchinvasion** [comment]
oh and yes buttons need fixing will do that

File: source/sass/_mixins.scss (hunk 2 above)
Comment (roidriscoll) [inline source/sass/_mixins.scss]: After a discussion with @stephenlloyd, we're going to stick with `$margin-lines` for the moment.

**Comment by roidriscoll** [comment]
@sjmog the scaling variables can be found in `vendor/bourbon/functions/_modular-scale.scss`

File: source/sass/_typography.scss [hunk 3]
```diff
@@ -15,50 +15,38 @@ h5,
...
   letter-spacing: 0.2em;
   text-transform: uppercase;
   text-rendering: optimizeLegibility;
   font-weight: 500;
   color: $brand-primary-color;
+  margin: 0 0 $base-line-height;
+  /*background-color: salmon;*/
 }
 
+/*p {*/
```
Comment (pitchinvasion) [inline source/sass/_typography.scss]: COMMENTS

File: source/sass/_variables.scss [hunk 4]
```diff
@@ -2,26 +2,28 @@
...
-$small-spacing: modular-scale(-2);
-$base-z-index: 0;
 $base-button-width: 10em;
 $mobile-nav-width: em(300);
 
+@mixin ma-header($step) {
+  $header-font-size: modular-scale($step);
+  $single_line: $base-line-height / $header-font-size;
+  $lines: ceil($base-font-size / $base-line-height);
+  /*$lines: ceil($header-font-size / em($base-line-height * $base-font-size));*/
```
Comment (pitchinvasion) [inline source/sass/_variables.scss]: COMMENTS

File: config.rb [hunk 5]
```diff
@@ -89,4 +92,7 @@
   set :run_analytics, true
 
   set :mixpanel_token, 'b839b30fbe0796d650ae20e7eae6d0d1'
+
+  ignore 'elements.html.haml'
+  ignore 'rhythm.html.haml'
```
Comment (pitchinvasion) [inline config.rb]: ignore typography too

**Pull request merged**